test/integration/component/test_network_offering.py

```diff
@@ -1449,14 +1449,16 @@ def test_create_network_with_snat(self):
 
         self.debug("Deploying VM in account: %s on the network %s" % (self.account.name, self.network.id))
         # Spawn an instance in that network
-        VirtualMachine.create(
+        self.vm1 = VirtualMachine.create(
             self.apiclient,
             self.services["virtual_machine"],
             accountid=self.account.name,
             domainid=self.account.domainid,
             serviceofferingid=self.service_offering.id,
             networkids=[str(self.network.id)]
         )
+        self.cleanup.append(self.vm1)
+        self.cleanup.append(self.network)
```

Comment:
this should move up and the tearDownClass method implementation should call super(..)
better clean the cleanup up ;)
cls._cleanup and self.cleanup